### PR TITLE
[3.4] Fields validation

### PR DESCRIPTION
**Pull request by AndrewVolosytnykhThingsboard**
Added validation according of database schema for back-end and front-end

### CODE REVIEW

File: ui-ngx/src/app/shared/components/entity/entity-subtype-autocomplete.component.ts [hunk 1]
```diff
@@ -122,6 +127,7 @@ export class EntitySubTypeAutocompleteComponent implements ControlValueAccessor,
         this.selectEntitySubtypeText = 'device.select-device-type';
         this.entitySubtypeText = 'device.device-type';
         this.entitySubtypeRequiredText = 'device.device-type-required';
+        this.entitySubtypeMaxLength = 'device.device-type-max-length';
```
Comment (vvlladd28) [inline ui-ngx/src/app/shared/components/entity/entity-subtype-autocomplete.component.ts:130]: Missing set value entitySubtypeMaxLength for EntityType.EDGE

File: ui-ngx/src/app/modules/home/components/entity/contact-based.component.ts [hunk 2]
```diff
@@ -14,15 +14,15 @@
...
-import { EntityTableConfig } from '@home/models/entity/entities-table-config.models';
+import {Store} from '@ngrx/store';
+import {AppState} from '@core/core.state';
+import {FormBuilder, FormGroup, ValidatorFn, Validators} from '@angular/forms';
+import {ContactBased} from '@shared/models/contact-based.model';
+import {AfterViewInit, Directive} from '@angular/core';
+import {POSTAL_CODE_PATTERNS} from '@home/models/contact.models';
+import {HasId} from '@shared/models/base-data';
+import {EntityComponent} from './entity.component';
+import {EntityTableConfig} from '@home/models/entity/entities-table-config.models';
```
Comment (vvlladd28) [inline ui-ngx/src/app/modules/home/components/entity/contact-based.component.ts:25]: Why change the code formatting?

**Review by vvlladd28 — APPROVED** [review]

**Comment by ViacheslavKlimov** [comment]
Pull request is reopened here: https://github.com/thingsboard/thingsboard/pull/5166

**Comment by YevhenBondarenko** [comment]
Pull request is reopened here: #5166

**Pull request closed without merging**